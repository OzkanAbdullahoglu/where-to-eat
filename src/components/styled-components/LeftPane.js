/* eslint-disable linebreak-style */
import styled from 'styled-components';

const setPaneContract = (conractRange, useProps) => (
  useProps.closed === true ? `translateX(${conractRange}px)` : 'translateX(0px)'
);

export const LeftPane = styled.div`
         width: 408px;
         position: absolute;
         top: 0;
         z-index: 3;
         opacity: 1;
         height: ${(props) => (props.closed === true ? 0 : '100%')};
         left: 0;
         -webkit-transform: ${(props) => setPaneContract(-408, props)};
         transform: ${(props) => setPaneContract(-408, props)};
         transition-property: -webkit-transform, transform, opacity;
         transition-duration: 0.2s;
         transition-timing-function: linear;
         box-shadow: ${(props) =>
    props.closed === true ? 'none' : '0 0 20px rgba(0, 0, 0, 0.3)'};
         background: #fff;
         @media (max-width: 414px) {
           width: 414px;
           -webkit-transform: ${(props) => setPaneContract(-414, props)};
           transform: ${(props) => setPaneContract(-414, props)}};
         }
         @media (max-width: 411px) {
           width: 411px;
           -webkit-transform: ${(props) => setPaneContract(-411, props)};
           transform: ${(props) => setPaneContract(-411, props)};
         }
         @media (max-width: 375px) {
           width: 375px;
           -webkit-transform: ${(props) => setPaneContract(-375, props)};
           transform: ${(props) => setPaneContract(-375, props)};
         }
         @media (max-width: 360px) {
           width: 360px;
           -webkit-transform: ${(props) => setPaneContract(-360, props)};
           transform: ${(props) => setPaneContract(-360, props)};
         }
         @media (max-width: 320px) {
           width: 320px;
           -webkit-transform: ${(props) => setPaneContract(-320, props)};
           transform: ${(props) => setPaneContract(-320, props)};
         }
       `;
