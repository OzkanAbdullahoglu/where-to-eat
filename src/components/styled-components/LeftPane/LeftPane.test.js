/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import LeftPane from './LeftPane';

describe('Left Pane Render', () => {
  const props = {
    closed: true,
  };

  it('renders left pane', () => {
    const leftPaneWrapper = shallow(<LeftPane {...props} />);
    expect(toJson(leftPaneWrapper)).toMatchSnapshot();
  });
  it('renders `closed` props', () => {
    const leftPaneWrapper = shallow(<LeftPane {...props} />);
    expect(leftPaneWrapper).toHaveStyleRule('transform', 'translateX(-408px)');
  });
});
