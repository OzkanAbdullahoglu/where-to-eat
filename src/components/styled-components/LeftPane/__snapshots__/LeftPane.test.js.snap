// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left Pane Render renders left pane 1`] = `
.c0 {
  width: 408px;
  position: absolute;
  top: 0;
  z-index: 3;
  opacity: 1;
  height: 0;
  left: 0;
  -webkit-transform: translateX(-408px);
  -webkit-transform: translateX(-408px);
  -ms-transform: translateX(-408px);
  transform: translateX(-408px);
  -webkit-transition-property: -webkit-transform,-webkit-transform,opacity;
  -webkit-transition-property: -webkit-transform,transform,opacity;
  transition-property: -webkit-transform,transform,opacity;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
  transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
  box-shadow: none;
  background: #fff;
}

@media (max-width:414px) {
  .c0 {
    width: 414px;
    -webkit-transform: translateX(-414px);
    -webkit-transform: translateX(-414px);
    -ms-transform: translateX(-414px);
    transform: translateX(-414px);
  }
}

@media (max-width:411px) {
  .c0 {
    width: 411px;
    -webkit-transform: translateX(-411px);
    -webkit-transform: translateX(-411px);
    -ms-transform: translateX(-411px);
    transform: translateX(-411px);
  }
}

@media (max-width:375px) {
  .c0 {
    width: 375px;
    -webkit-transform: translateX(-375px);
    -webkit-transform: translateX(-375px);
    -ms-transform: translateX(-375px);
    transform: translateX(-375px);
  }
}

@media (max-width:360px) {
  .c0 {
    width: 360px;
    -webkit-transform: translateX(-360px);
    -webkit-transform: translateX(-360px);
    -ms-transform: translateX(-360px);
    transform: translateX(-360px);
  }
}

@media (max-width:320px) {
  .c0 {
    width: 320px;
    -webkit-transform: translateX(-320px);
    -webkit-transform: translateX(-320px);
    -ms-transform: translateX(-320px);
    transform: translateX(-320px);
  }
}

<div
  className="c0"
/>
`;
